libbeat/processors/add_cloud_metadata/docs/add_cloud_metadata.asciidoc

```diff
@@ -50,6 +50,10 @@ List of names the `providers` setting supports:
 
 - "alibaba", or "ecs" for the Alibaba Cloud provider (disabled by default).
 - "azure" for Azure Virtual Machine (enabled by default).
+   If the Virtual Machine is part of an AKS managed cluster, the fields
+   `orchestrator.cluster.name` and `orchestrator.cluster.id` can only be
+   retrieved if "TENANT_ID", "CLIENT_ID" and "CLIENT_SECRET" environment
+   variables are set.
```

Comment:
Using `azidentity.NewDefaultAzureCredential(nil)` as a fallback when there ir no environment variables is a good call. 

`azidentity.NewDefaultAzureCredential(nil)` gives more options to the users that want and have the knowledge to use different authentication mechanisms like the Workload Identity [we experimented](https://github.com/zmoog/public-notes/issues/72) with.

What about mentioning this opportunity in the docs?